sphinx/util/inspect.py

```diff
@@ -158,3 +158,26 @@ def is_builtin_class_method(obj, attr_name):
     if not hasattr(builtins, safe_getattr(cls, '__name__', '')):
         return False
     return getattr(builtins, safe_getattr(cls, '__name__', '')) is cls
+
+
+def dumb_signature(functional_object):
+    src = inspect.getsourcelines(functional_object)[0]
+
+    for i, l in enumerate(src):
+        if 'def' in l:
+            break
+    acc = ''
+    for l in src[i:]:
+        acc += l.strip()
+        if '):' in acc:
+            break
+
+    r = re.compile(r'\((.*)\)')
```

Comment:
This breaks for any kind of tuple or function call inside the signature, eg. ```def foo(a=(1,2)):``` or ```def bar(b=get_default_for_b())```.